### DISCUSSION TITLE
Deta Base python SDK: how can I delete every object inside a base?

### DISCUSSION THREAD

**Discussion by robsannaa** [deta/cloud-docs#588] [Help] [answered]
As the title suggests, I would like to delete every record inside a base without having their IDs. Is there a way to do it?

**Comment by sudo-subg** [comment, marked as answer]
As of now there is no way to delete every record inside a deta base without having there keys. The only workaround is getting all the keys with [fetch everything](https://docs.deta.sh/docs/base/sdk#fetch-all-items-1) and then delete everything one by one in a loop.

**Comment by el-garro** [comment]
Atm there's nothing, you could technically use the same endpoint the UI uses to delete the base, but that requires aditional login instead of just project key.

**Comment by jnsougata** [comment]
> As the title suggests, I would like to delete every record inside a base without having their IDs. Is there a way to do it?

do you wanna delete the base instead?
